.changeset/purple-flies-end.md

```diff
@@ -0,0 +1,23 @@
+---
+'@razorpay/blade': minor
+---
+
+**feat: Added internationalization in Amount component via i18nify.
+References**
+
+- **i18nify-js:** https://www.npmjs.com/package/@razorpay/i18nify-js
+- **i18nify-react:** https://www.npmjs.com/package/@razorpay/i18nify-react
+
+**What changes ?**
+
+1. The `<Amount />` component will now automatically format numbers based on the user's browser locale. For example, `<Amount value={123456.789} currency="INR">` will render `₹1,23,456.79` for browsers with the `en-IN` default locale, whereas it will render `₹123,456.79` for browsers with the `en-US` locale.
+
+2. If you want to enable users to change the locale of your page, add the `@razorpay/i18nify-react` package and wrap your app inside the `I18nProvider`. Utilize the `setI18nState` utility to modify the locale. For more details, please refer to the [documentation](https://www.npmjs.com/package/@razorpay/i18nify-react).
+
+3. Additionally, if you prefer to maintain a fixed locale for your page and amount component, enclose your app within `<I18nProvider data={{locale: 'locale-you-want'}}>..`. For more details, please refer to the [documentation](https://www.npmjs.com/package/@razorpay/i18nify-react).
+
+**How to update ?**
+
+1. Install i18nify peer dependency `yarn add @razorpay/i18nify-js`
+2. _[Optional]_: Install i18nify-react peer dependency to manage state effectively `yarn add @razorpay/i18nify-react`
```

Comment:
We were discussing about breaking change internally. We decided to have this as minor release and add its changelog into v11 https://github.com/razorpay/blade/blob/master/packages/blade/upgrade-v11.md#amount because we very recently had a breaking release which people are yet to migrate to.

I think we'll have `@razorpay/i18nify-react` as mandatory dependency and ask consumers to add a `I18nProvider` on their end with expected locale. 

A bit risky to rely on browser locale for existing Amount components since our default so far has been en-IN behaviour so things might break for product if something like this changes - https://github.com/razorpay/blade/pull/2003/files#r1482641498

Reply:
@saurabhdaware What are the next steps here ? 
I think `@razorpay/i18nify-react` should be just added as mandatory step in migration guide to v11 but shouldn't be added as a mandatory peer dependecy for new users.

Reply:
> I think @razorpay/i18nify-react should be just added as mandatory step in migration guide to v11 but shouldn't be added as a mandatory peer dependecy for new users.

Yes makes sense

Reply:
> shouldn't be added as a mandatory peer dependecy for new users.

regarding this, should we explicitly mark `@razorpay/i18nify-react` as optional via peerDependenciesMeta?